sycl/include/CL/sycl/intel/fpga_utils.hpp

```diff
@@ -0,0 +1,34 @@
+//==------------- fpga_utils.hpp --- SYCL FPGA Reg Extensions --------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/stl.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace intel {
+
+template <template <int32_t> class Type, class T>
+struct MatchType : std::is_same<Type<T::value>, T> {};
+
+template <template <int32_t> class Type, class... T> struct GetValue {
+  enum { value = Type<0>::default_value };
+};
+
+template <template <int32_t> class Type, class T1, class... T>
+struct GetValue<Type, T1, T...> {
+  enum {
+    value = std::conditional<MatchType<Type, T1>::value, T1,
+                             GetValue<Type, T...>>::type::value
```

Comment:
Perhaps you could have a lighter implementation by using a more modern design (using `std::tuple`?) but it is not important.
At least a simpler `static auto constexpr value =` too?

Reply:
I went with the second option for now. I will consider redesigning using `std::tuple` in another change :)

Reply:
Keep it as is, since we might come with a different API in the future.
At least your work gives me some ideas to try with some Xilinx extensions @aisoard @Ralender 
Something like in your case
```c++
auto constexpr PrefetchingLSU = cl::sycl::intel::lsu(prefetch(true), statically_coalesce(false));
int X = PrefetchingLSU.load(input_ptr); // int X = input_ptr[0]
```
where `prefetch` and `statically_coalesce` are found by ADL to shorten the syntax.
To experiment whether it is possible...